### PR TITLE
fix character access in autograph

### LINKED ISSUE #20703: Character access is not supported in tf.contrib.autograph
For example this is not working and it returns an error
```
@autograph.convert()
def is_hashtag(x):
  if x[0] == "#":
    return True
  return False

with tf.Graph().as_default():  
  with tf.Session() as sess:
    print(sess.run(is_hashtag(tf.constant("#asdf"))))
```

### PR DESCRIPTION
**Pull request by ariwaranosai**
fix #20703 by extending the tensor index operator.

### CODE REVIEW

**Comment by ariwaranosai** [comment]
@mdanatg  could you take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @mdanatg: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 16 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/contrib/autograph/operators/slices_test.py [hunk 1]
```diff
@@ -46,6 +46,19 @@ def test_get_item_tensor_list(self):
     with self.test_session() as sess:
       self.assertAllEqual(sess.run(t), [3, 4])
 
+  def test_get_item_tensor_string(self):
+    initial_str = constant_op.constant("abcd")
```
Comment (mdanatg) [inline tensorflow/contrib/autograph/operators/slices_test.py]: For consistency, please use single quotation marks for single-line strings.

**Comment by mdanatg** [comment]
This is great! Sorry for the delay, I just got back from vacation.

**Review by mdanatg — DISMISSED** [review]

**Comment by ariwaranosai** [comment]
@mdanatg you are right, fixed.

**Review by mdanatg — DISMISSED** [review]

**Comment by ariwaranosai** [comment]
I just realized this will raise error for tensor of string. This should only apply for string with rank 0.
@mdanatg

File: tensorflow/contrib/autograph/operators/slices.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def get_item(target, i, opts):
   elif tensor_util.is_tensor(target):
     if target.dtype == dtypes.variant:
       return _tf_tensor_list_get_item(target, i, opts)
-    if target.dtype == dtypes.string:
+    elif target.dtype == dtypes.string and target.get_shape() == (): # target is string with rank 0
```
Comment (mdanatg) [inline tensorflow/contrib/autograph/operators/slices.py]: More idiomatic: `target.shape.ndims == 0`

https://www.tensorflow.org/api_docs/python/tf/TensorShape#ndims

**Review by mdanatg — APPROVED** [review]

**Comment by ariwaranosai** [comment]
Jenkins test this please.

**Pull request merged**